From 3fe963510d04e696bbc1c12cbea475b2ca8875f7 Thu, 12 Nov 2020 20:04:42 +0100
From: Andrew Branson <andrew.branson@jolla.com>
Date: Thu, 12 Nov 2020 20:03:34 +0100
Subject: [PATCH] [sailfishos][gecko] Force recycling of gmp-droid instances. JB#51730


Android codecs hog a lot of frames, so creating a new one during playback when 
the stream switches will cause lag and playback failure. These codecs are
capable of switching streams internally, so flag gmp-droid as supporting 
decoder recycling.

diff --git a/dom/media/platforms/agnostic/gmp/GMPVideoDecoder.h b/dom/media/platforms/agnostic/gmp/GMPVideoDecoder.h
index e9f0843..e1d15bd 100644
--- a/dom/media/platforms/agnostic/gmp/GMPVideoDecoder.h
+++ b/dom/media/platforms/agnostic/gmp/GMPVideoDecoder.h
@@ -42,6 +42,9 @@
   nsCString GetDescriptionName() const override {
     return NS_LITERAL_CSTRING("gmp video decoder");
   }
+  bool SupportDecoderRecycling() const override {
+      return mGMP->GetDisplayName().EqualsLiteral("gmp-droid");
+  }
   ConversionRequired NeedsConversion() const override {
     return mConvertToAnnexB ? ConversionRequired::kNeedAnnexB
                             : ConversionRequired::kNeedAVCC;
